# Author: Stefan Harfst, HPC-UniOldenburg
# Date: 23.04.2020
# adding fixes (namely namespace and include guard) for strict compilers 


--- fsl/src/DiscreteOpt/DiscreteCostFunction.h.orig	2020-04-02 17:04:51.483993000 +0200
+++ fsl/src/DiscreteOpt/DiscreteCostFunction.h	2020-04-02 17:06:03.649326000 +0200
@@ -65,6 +65,8 @@
     University Innovation ("OUI"), the technology transfer company of the
     University, to negotiate a licence. Contact details are:
     fsl@innovation.ox.ac.uk quoting Reference Project 9564, FSL.*/
+#ifndef _DISCRETECOSTFUNCTION_H_
+#define _DISCRETECOSTFUNCTION_H_ 
 #include <vector>
 #include "similarities.h"
 
@@ -718,3 +720,4 @@
 
 
 }
+#endif
--- fsl/src/DiscreteOpt/DiscreteGroupCostFunction.cpp.orig	2019-11-19 11:46:56.000000000 +0100
+++ fsl/src/DiscreteOpt/DiscreteGroupCostFunction.cpp	2020-04-02 16:08:40.714526000 +0200
@@ -68,6 +68,10 @@
 #include "DiscreteGroupCostFunction.h"
 #include <algorithm>
 
+// adding fix for strict compilers
+// (see https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=FSL;3909b571.1907)
+using namespace DISCRETEOPT;
+
 namespace DISCRETEOPTHOCR{
   
   //====================================================================================================================//
--- fsl/src/DiscreteOpt/DiscreteGroupCostFunction.h.orig	2019-11-19 11:46:56.000000000 +0100
+++ fsl/src/DiscreteOpt/DiscreteGroupCostFunction.h	2020-04-02 16:08:40.733656000 +0200
@@ -73,6 +73,10 @@
 //#include "tbb/tick_count.h"
 //using namespace tbb;
 
+// adding fix for strict compilers 
+// (see https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=FSL;3909b571.1907)
+using namespace DISCRETEOPT;
+
 typedef map<string, boost::variant<int, string, double, float, bool> > myparam;
 typedef pair<string, boost::variant<int, string, double, float, bool> > parameterPair;
 
--- fsl/src/DiscreteOpt/DiscreteGroupModel.cpp.orig	2019-11-19 11:46:56.000000000 +0100
+++ fsl/src/DiscreteOpt/DiscreteGroupModel.cpp	2020-04-02 16:08:40.752795000 +0200
@@ -69,6 +69,9 @@
 #include <algorithm>
 #include <iostream>
 
+// adding fix for strict compilers
+// (see https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=FSL;3909b571.1907)
+using namespace DISCRETEOPT;
 
 namespace DISCRETEOPTHOCR{
 
--- fsl/src/DiscreteOpt/DiscreteGroupModel.h.orig	2019-11-19 11:46:56.000000000 +0100
+++ fsl/src/DiscreteOpt/DiscreteGroupModel.h	2020-04-02 16:08:40.771535000 +0200
@@ -72,7 +72,9 @@
 #include "DiscreteModel.h"
 #include <vector>
 
- 
+// adding fix for strict compilers
+// (see https://www.jiscmail.ac.uk/cgi-bin/webadmin?A2=FSL;3909b571.1907)
+using namespace DISCRETEOPT;
 
 namespace DISCRETEOPTHOCR{
  
--- fsl/src/DiscreteOpt/DiscreteModel.h.orig	2020-04-02 17:07:28.284511000 +0200
+++ fsl/src/DiscreteOpt/DiscreteModel.h	2020-04-02 17:08:24.121019000 +0200
@@ -65,6 +65,8 @@
     University Innovation ("OUI"), the technology transfer company of the
     University, to negotiate a licence. Contact details are:
     fsl@innovation.ox.ac.uk quoting Reference Project 9564, FSL.*/
+#ifndef _DISCRETEMODEL_H_
+#define _DISCRETEMODEL_H_ 
 #include "DiscreteCostFunction.h"
 #include <vector>
 
@@ -686,3 +688,4 @@
 
 
 }
+#endif
